include/ignition/rendering/ShaderParam.hh

```diff
@@ -64,29 +72,50 @@ namespace ignition
       /// \return Type of this parameter
       public: ParamType Type() const;
 
-      /// \brief Set from another ShaderParam.
-      /// \param[in] _other Another ShaderParam.
-      /// \return Reference to this ShaderParam.
+      /// \brief Get the element count of this parameter's buffer
+      /// \return Count of elements in this parameter's buffer
+      public: uint32_t Count() const;
+
+      /// \brief Set from another ShaderParam
+      /// \param[in] _other Another ShaderParam
+      /// \return Reference to this ShaderParam
       public: ShaderParam &operator=(const ShaderParam &_other);
 
-      /// \brief Set this to be a float param
-      /// \param[in] _value Value to set this param to.
+      /// \brief Set this to be a float parameter
+      /// \param[in] _value Value to set this parameter to
       public: void operator=(const float _value);
 
-      /// \brief Set this to be an integer param;
-      /// \param[in] _value Value to set this param to.
+      /// \brief Set this to be an integer parameter
+      /// \param[in] _value Value to set this parameter to
       public: void operator=(const int _value);
 
+      /// \brief Set this to be a word buffer parameter
+      /// \param[in] _count Number of elements in the word buffer
+      public: void InitializeWordBuffer(const uint32_t _count);
+
+      /// \brief Copy a buffer to this parameter
+      /// \param[in] _floatBuffer Source buffer to copy from
+      public: void UpdateWordBuffer(float *_floatBuffer);
+
+      /// \brief Copy a buffer to this parameter
+      /// \param[in] _intBuffer Source buffer to copy from
+      public: void UpdateWordBuffer(int *_intBuffer);
+
       /// \brief Get the value of this parameter if it is a float
       /// \param[out] _value variable the value will be copied to
-      /// \return true if the param is the expected type
+      /// \return true if the parameter is the expected type
       public: bool Value(float *_value) const;
 
       /// \brief Get the value of this parameter if it is an int
       /// \param[out] _value variable the value will be copied to
-      /// \return true if the param is the expected type
+      /// \return true if the parameter is the expected type
       public: bool Value(int *_value) const;
 
+      /// \brief Get the value of this parameter if it is a word buffer
+      /// \param[out] _wordBuffer variable the value will be copied to
+      /// \return true if the parameter is the expected type
+      public: bool WordBuffer(std::shared_ptr<void> *_wordBuffer) const;
```

Comment:
use raw pointer?

Reply:
I was thinking about using a reference instead, but I wanted to match the other existing methods.

```
      /// \brief Get the value of this parameter if it is a float
      /// \param[out] _value variable the value will be copied to
      /// \return true if the parameter is the expected type
      public: bool Value(float *_value) const;

      /// \brief Get the value of this parameter if it is an int
      /// \param[out] _value variable the value will be copied to
      /// \return true if the parameter is the expected type
      public: bool Value(int *_value) const;

      /// \brief Get the value of this parameter if it is a word buffer
      /// \param[out] _wordBuffer variable the value will be copied to
      /// \return true if the parameter is the expected type
      public: bool WordBuffer(std::shared_ptr<void> *_wordBuffer) const;
```

In this case, should I update them all to `shared_ptr`s?

Reply:
we won't be able to change APIs for a released version. The changes would need to go into Ignition F / ign-rendering6. We would deprecate these functions and add new one in the `main` branch. 

You can however make changes to `WordBuffer` you just added since it's a new function.